src/main/java/edu/cornell/kfs/module/purap/document/service/impl/CuB2BShoppingServiceImpl.java

```diff
@@ -183,15 +190,48 @@ public List createRequisitionsFromCxml(B2BShoppingCart message, Person user) thr
 
             req.fixItemReferences();
 
-            // save requisition to database
-            purapService.saveDocumentNoValidation(req);
+            if (checkRequisitionAccountsAreUnique(req)) {
+                // save requisition to database
+                purapService.saveDocumentNoValidation(req);
 
-            // add requisition to List
-            requisitions.add(req);
+                // add requisition to List
+                requisitions.add(req);
+            }
+        }
+
+        if (Collections.isEmpty(requisitions)) {
+            throw new B2BShoppingException("Invalid e-shop Request. No requisitions were created.");
         }
+
         return requisitions;
     }
 
+    public boolean checkRequisitionAccountsAreUnique(RequisitionDocument requisitionDocument) {
+
+        List<RequisitionItem> requisitionItems = requisitionDocument.getItems();
+        boolean requisitionAccountsUnique = true;
+        for (RequisitionItem requisitionItem : requisitionItems) {
+
+            PurchasingAccountsPayableUniqueAccountingStringsValidation uniqueRequisitionAccountValidator =
```

Comment:
I would move PurchasingAccountsPayableUniqueAccountingStringsValidation to a member variable, and use getters and setters.  In the getter, if the member variable is NULL, us springcontext to set it.  That way, it is easier to test